Private sector data dumps

mkangia opened the pull request:
[Specs](https://docs.google.com/spreadsheets/d/1t6cd-VPy6p8EOEhQJD15IbULU0EJ05ALQ0tcdfx6ng8/edit#gid=1299017227)

All of the dumps are done with few clarifications needed and is good to review. mock dumps for almost all of the them with GS for review..

mkangia committed:
Add data dumps for private person cases [skip ci]

mkangia committed:
Add data dumps for private episode details [skip ci]

mkangia committed:
fix mapping csv files [skip ci]

mkangia committed:
update fields for person case data dump [skip ci]

mkangia committed:
Add data dumps for private prior diagnostics [skip ci]

mkangia committed:
Add data dumps for private drug voucher details [skip ci]

mkangia committed:
Add data dumps for private payments [skip ci]

mkangia committed:
Add data dumps for private record adherence [skip ci]

mkangia committed:
Add data dumps for private diagnostic voucher details [skip ci]

mkangia committed:
Add data dumps for private adherence summary [skip ci]

mkangia committed:
Add data dumps for private data exchanged with bets [skip ci]

mkangia committed:
Fix adherence summary [skip ci]

mkangia committed:
Fix data exhanged with bets [skip ci]

mkangia committed:
remove redundant reference expected for diagnostic voucher details [skip ci]

mkangia committed:
remove redundant import [skip ci]

mkangia committed:
fix minor fixes [skip ci]

mkangia committed:
handle case when voucher not found causing exception with key voucher[skip ci]

mkangia committed:
fix naming and input file name[skip ci]

mkangia committed:
ignore other voucher types for voucher [skip ci]

mkangia committed:
fix issue with record adherence script[skip ci]

mkangia committed:
lint [skip ci]

mkangia committed:
fetch information from multiple episodes for prior_diagnostics and diagnostic_voucher_details [skip ci]

mkangia committed:
remove calls for "get_episode" and fix expected cases [skip ci]

mkangia committed:
add check for vouchers [skip ci]

esoergel commented on custom/enikshay/management/commands/data_dumps_private_payments.py:
```diff
@@ -0,0 +1,144 @@
+
+    def get_case_ids_query(self, case_type):
+        """
+        ToDo: add date_fulfilled check in ES case ids query itself
```
Do you still plan to address this `ToDo`?  I imagine you could also check the case state to see whether it's been fulfilled.

mkangia replied:
i already added this on the case in `include_case_in_dump`.. but adding this in ES would help smaller bunch of ids..I just was not clear how to add this in ES. checking if a case property is not null or is present. Can you suggest how and if this can be added via ES query?

esoergel replied:
Seems fine to skip.  It's doable, would just require making a new case_search filter.

I think it would basically be [`case_property_filter`](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/es/case_search.py#L106-L113), but with the `value` line using `filters.empty` instead of `filters.term`.

esoergel commented on custom/enikshay/management/commands/data_dumps_private_person_case.py:
```diff
@@ -0,0 +1,79 @@
+                if location:
+                    return location.name
+                else:
+                    return "FO Location not found with id: %s" % fo_id
```
nit: totally fine in a one-off context, but in general I'd write this sort of thing as

```python
try:
    return SQLLocation.objects.get(location_id=fo_id).name
except SQLLocation.DoesNotExist:
    return "FO Location not found with id: %s" % fo_id
```
Just because that seems more aligned with how the Django ORM is structured.  `get_or_None` makes sense if you'll be returning the result and/or may need a `None` value.

esoergel commented on custom/enikshay/management/commands/data_dumps_private_record_adherence.py:
```diff
@@ -0,0 +1,95 @@
+        return (self.case_search_instance
+                .case_type(case_type)
+                .case_property_query(ENROLLED_IN_PRIVATE, 'true', clause=queries.MUST)
+                # .case_property_filter("episode_type", ["confirmed_tb", "confirmed_drtb"])
```
Should this filter be applied?

esoergel reviewed (approved):
Looks good!  Is this the last batch?

mkangia commented on custom/enikshay/management/commands/data_dumps_private_data_exchanged_with_bets.py:
```diff
@@ -0,0 +1,136 @@
+
+
+class Command(BaseDataDump):
+    """ 13. data dumps for private data exchanged with BETS
```
thanks @esoergel . Will flag that to the GS team.

mkangia committed:
review fixes [skip ci]

fix overwriting var in list comprehension
better way to fetch location name

mkangia committed:
ignore cases that have cases missing and log them for inspection [skip ci]

mkangia committed:
Merge branch 'master' into mk-private-data-dumps

mkangia committed:
fix list comprehension [skip ci]

mkangia committed:
add check for location for all dumps [skip ci]

mkangia committed:
fix check for no cases [skip ci]

mkangia committed:
update column names and pull location name instead for lab id and name [skip ci]

mkangia committed:
add mapping for org ids [skip ci]

mkangia committed:
diagnostic voucher details fixes [skip ci]

1. fix case property
2. display username and full name
3. display location name

mkangia committed:
update "Organisation" column for all sheets instead [skip ci]

mkangia committed:
private person case fixes [skip ci]

mkangia committed:
fix column Facility/Provider assigned to Name - ID [skip ci]

mkangia committed:
Raise instead of returning string response [skip ci]

mkangia committed:
drug voucher details fixes [skip ci]

mkangia committed:
payments fixes [skip ci]

mkangia committed:
fix modified at and modified by columns [skip ci]

mkangia committed:
fix Date of Episode creation [skip ci]

mkangia committed:
raise if no location selected and remove block since its already a name [skip ci]

mkangia committed:
fix Organisations column [skip ci]

mkangia committed:
add Date of episode creation [skip ci]

mkangia committed:
fix error with list comprehension with datetime [skip ci]

mkangia committed:
fixes for data exchange with BETS [skip ci]

mkangia committed:
adherence summary fixes [skip ci]

mkangia committed:
share failure reason only when present and convert datetime to str [skip ci]

mkangia committed:
import division[skip ci]

mkangia committed:
lint [skip ci]

mkangia committed:
lint [skip ci]

mkangia committed:
fix extra space after current_episode_type [skip ci]

mkangia committed:
rescue missing cases [skip ci]

mkangia committed:
avoid printing, just skip [skip ci]

mkangia committed:
return username instead of name [skip ci]

mkangia committed:
return message if date start after date recorded [skip ci]

mkangia committed:
rescue if case missing [skip ci]

mkangia committed:
ignore person specific person case ids [skip ci]

mkangia committed:
add Organization column custom logic [skip ci]

mkangia committed:
Revisit record adherence and break it down to multiple files in the code directory itself [skip ci]

mkangia committed:
person dump fixes [skip ci]

mkangia committed:
fix voucher type for bets and payments and add custom columns [skip ci]

mkangia committed:
fix drug voucher details [skip ci]

mkangia committed:
add Organisation custom column and fix calculation logic [skip ci]

mkangia committed:
fix hierarchy for private payments and bets [skip ci]

mkangia committed:
fix organization column for bets and issued product name [skip ci]

mkangia committed:
add notes [skip ci]

mkangia committed:
episode details fixes [skip ci]

mkangia committed:
data exchanged with bets fixes [skip ci]
